### PR TITLE
Add support for Header and ROS Time

### LINKED ISSUE #64: 03_slow_consumer.py example produces Header with floating-point time fields
Context: https://github.com/gramaziokohler/roslibpy/issues/57#issuecomment-685410197

[This example](https://github.com/gramaziokohler/roslibpy/blob/51a1a06643efb0690cfb7495c08160fc814d278c/docs/examples/03_slow_consumer.py#L21) produces a floating-point `nsecs` value, rather than integer.

## Expected Result

The message to publish without error.

## Actual Result

Error message (from what I assume is the rosbridge):

```
[ERROR] [1599032244.517260]: [Client 0] [id: publish:/serial/turret_aim_feedback:161] publish: <class 'struct.error'>: 'required argument is not an integer' when writing 'header
```

## Reproduction Steps

Above sample should do it.

## System Information

Ubuntu 20.04; ROS noetic.

### PR DESCRIPTION
**Pull request by gonzalocasas**
Closes #64

Added support for `Header` and `Time` (ROS) data types mimicking closely  its `rospy` counter-parts including the method naming to convert back and forth between `Time` and epoch time in floating point (Python's). Together with this, I added an (internal) class to handle msg encoding, eventually, this could be made public, allowing the user to define its own encoder, and add the decoding part as well.

### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [x] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I added a line to the `CHANGELOG.rst` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [x] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [x] I ran lint on my computer and there are no errors (i.e. `invoke check`).
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by gonzalocasas** [comment]
This is ready for review :)

File: tests/test_core.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    t = Time.from_sec(REF_FLOAT_SECS_TIME)
+    assert t.to_nsec() == 1610122759677661895
+
+
+def test_to_sec():
+    t = Time.from_sec(REF_FLOAT_SECS_TIME)
+    assert t.to_sec() == REF_FLOAT_SECS_TIME
+
+
+def test_is_zero():
```
Comment (beverlylytle) [inline tests/test_core.py:23]: I would also add some negative tests here, eg `assert not Time(1, 0).is_zero()`.

File: src/roslibpy/core.py [hunk 2]
```diff
@@ -30,6 +31,59 @@ def __init__(self, values=None):
...
+
+    def is_zero(self):
+        """Return ``True`` if zero (secs and nsecs) otherwise ``False``."""
+        return self.data['secs'] == 0 and self.data['nsecs'] == 0
+
+    def to_nsec(self):
+        """Return time as nanoseconds from epoch."""
+        stamp_secs = self.data['secs']
+        stamp_nsecs = self.data['nsecs']
+        return stamp_secs * 1000000000 + stamp_nsecs
```
Comment (beverlylytle) [inline src/roslibpy/core.py:68]: I find `1000000000` hard to read and would prefer `1e9`, but then it becomes a float. So, as you prefer.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      - master
+
+jobs:
+  build-cpython:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        name: [
+          "ubuntu-py37",
+          "ubuntu-py36",
```
Comment (beverlylytle) [inline .github/workflows/build.yml:18]: No more 2.7?  oh, so sad.  But should that be reflected in `setup.py`?

**Review by beverlylytle — APPROVED** [review]
LGTM!  There are a couple of minor comments, but addressing them is up to you.

File: .github/workflows/build.yml (hunk 3 above)
Comment (gonzalocasas) [reply .github/workflows/build.yml:18]: done at https://github.com/gramaziokohler/roslibpy/pull/72/commits/f4d9cc16181c1678f403d53381e7446fcde71534

File: src/roslibpy/core.py (hunk 2 above)
Comment (gonzalocasas) [reply src/roslibpy/core.py]: done at https://github.com/gramaziokohler/roslibpy/pull/72/commits/d3f0ab649f4db8b2a6362e07260a6474a0332c08

File: tests/test_core.py (hunk 1 above)
Comment (gonzalocasas) [reply tests/test_core.py:23]: added on https://github.com/gramaziokohler/roslibpy/pull/72/commits/a08dea94559d809a9b5edaa155542ac26e58e81f

**Pull request merged**